metric_config_parser/config.py

```diff
@@ -288,125 +302,44 @@ def from_github_repo(
         depth: Optional[int] = None,
     ) -> "ConfigCollection":
         """Pull in external config files."""
-        # download files to tmp directory
-        with TemporaryDirectory() as tmp_dir:
-            if repo_url is not None and Path(repo_url).exists() and Path(repo_url).is_dir():
-                if path:
-                    repo = Repo(repo_url)
-                else:
-                    repo_path = Path(repo_url)
-                    dir_path = Path()
-                    depth = depth or 3
-
-                    # check if parent folder is a repository until search depth is exceeded
-                    while depth > 0:
-                        try:
-                            repo = Repo(repo_path)
-                        except InvalidGitRepositoryError:
-                            dir_path = Path(repo_path.name) / dir_path
-                            repo_path = repo_path.parent
-                            depth -= 1
-                        else:
-                            depth = 0
-
-                    repo = Repo(repo_path)
-                    path = str(dir_path)
-                    repo_url = str(repo_path)
-
-                tmp_dir = Path(repo_url)
+        # download files to a persisted tmp directory
+        tmp_dir = Path(tempfile.mkdtemp())
```

Comment:
using `tempfile.mkdtemp()` instead of `TemporaryDirectory()` otherwise the temporary directory including the repo get deleted as soon as it goes out of scope. Keeping the repo around is just more efficient rather than downloading it every time

Reply:
Good point, I'll add it to the class destructor